services/utils/akka/src/main/java/org/eclipse/ditto/services/utils/akka/controlflow/DittoFlowEnhancement.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.ditto.services.utils.akka.controlflow;
+
+
+import java.time.Duration;
+
+import javax.annotation.Nullable;
+
+import org.eclipse.ditto.services.utils.metrics.instruments.timer.PreparedTimer;
+
+import akka.actor.ActorRef;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Flow;
+import akka.stream.javadsl.Sink;
+
+public final class DittoFlowEnhancement<I, O, M> {
```

Comment:
Maybe i miss something, but i don't really get the benefit of this class, besides of the sexy currying. The naming and the docs also don't really give useful insights on that (`...enhance a given flow with some add-on behaviour`). Instead, this class complect multiple independent pieces. So i would favor to have self-contained universal building-blocks (`TimeoutFlow`, `TimeMeasuringFlow`,..) instead of something new.

Reply:
It's meant to be a single source of all "add-ons" for flows. Without this class a developer would need to be aware of every single implementation (TimeoutFlow, TimeMeasuringFlow, ...).